services/blockchain-indexer/shared/indexer/transactionProcessor/dpos/registerDelegate.js

```diff
@@ -17,28 +17,37 @@ const {
 	Logger,
 	MySQL: { getTableInstance },
 } = require('lisk-service-framework');
+
+const { getBase32AddressFromPublicKey } = require('../../../utils/accountUtils');
 const config = require('../../../../config');
 
 const logger = Logger();
 
 const MYSQL_ENDPOINT = config.endpoints.mysql;
-const transactionsIndexSchema = require('../../../database/schema/transactions');
+const accountsIndexSchema = require('../../../database/schema/accounts');
 
-const getTransactionsIndex = () => getTableInstance('transactions', transactionsIndexSchema, MYSQL_ENDPOINT);
+const getAccountsIndex = () => getTableInstance('accounts', accountsIndexSchema, MYSQL_ENDPOINT);
 
 // Command specific constants
 const commandID = 0;
 const commandName = 'registerDelegate';
 
 // eslint-disable-next-line no-unused-vars
 const processTransaction = async (blockHeader, tx, dbTrx) => {
-	const transactionsDB = await getTransactionsIndex();
-
-	// TODO: Implement logic
-
-	logger.trace(`Indexing transaction ${tx.id} contained in block at height ${tx.height}`);
-	await transactionsDB.upsert(tx, dbTrx);
-	logger.debug(`Indexed transaction ${tx.id} contained in block at height ${tx.height}`);
+	const accountsDB = await getAccountsIndex();
+
+	const account = {
+		address: getBase32AddressFromPublicKey(tx.senderPublicKey),
+		publicKey: tx.senderPublicKey,
+		name: tx.params.name,
+		generatorKey: tx.params.generatorKey,
+		blsKey: tx.params.blsKey,
+		proofOfPosession: tx.params.proofOfPosession,
+	};
+
+	logger.trace(`Updating account index for the account with address ${account.address}`);
+	await accountsDB.upsert(account, dbTrx);
```

Comment:
Please adjust the logic after modifying the `accounts` and creating the `generators` tables.